### PR TITLE
[MAINTENANCE] expectations linting & bug fixes

### PR DESCRIPTION
**Pull request by Kilo59**
## Changes proposed in this pull request:
- enable `flake8` linting for `expectations` (excluding `metrics` sub-package)
- [x] fix undefined `num_to_str()`
- [ ] ~fix undefined `params_with_json_schema`~ - created bug-ticket

`render.util.num_to_str()` is the only `num_to_str()` function in our codebase, so I'm assuming this was the intended method.
https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/render/util.py#L22

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code

### CODE REVIEW

File: great_expectations/expectations/row_conditions.py [hunk 1]
```diff
@@ -148,7 +157,7 @@ def parse_condition_to_spark(row_condition: str) -> "pyspark.sql.Column":
 
 def parse_condition_to_sqlalchemy(
     row_condition: str,
-) -> "sqlalchemy.sql.expression.ColumnElement":
+) -> ColumnElement | None:
```
Comment (Kilo59) [inline great_expectations/expectations/row_conditions.py:160]: save the `None` for typing update

**Comment by ghost** [comment]
<details>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=6739&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

  - Make big changes easier — review code in small groups of related files
  - Know where to start — see the whole change at a glance
  - Take a code tour — explore the change with an interactive tour
  - Make comments and review — all fully sync’ed with github

    [Try it now!](https://app.codesee.io/r/reviews?pr=6739&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)
</details>

[Review these changes using an interactive CodeSee Map](https://app.codesee.io/r/reviews?pr=6739&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
CodeSee Map Legend

File: great_expectations/expectations/core/expect_select_column_values_to_be_unique_within_record.py [hunk 2]
```diff
@@ -177,7 +175,7 @@ def _prescriptive_renderer(
         )
 
         if params["mostly"] is not None and params["mostly"] < 1.0:
-            params_with_json_schema["mostly_pct"]["value"] = num_to_str(
+            params_with_json_schema["mostly_pct"]["value"] = num_to_str(  # FIXME
```
Comment (Kilo59) [inline great_expectations/expectations/core/expect_select_column_values_to_be_unique_within_record.py]: This appears broken.

`params_with_json_schema` is not defined

File: great_expectations/expectations/regex_based_column_map_expectation.py [hunk 3]
```diff
@@ -220,7 +220,7 @@ def _prescriptive_renderer(
         else:
             template_str = "values must match this regular expression: $regex"
             if params["mostly"] is not None:
-                params["mostly_pct"] = num_to_str(
+                params["mostly_pct"] = num_to_str(  # FIXME
```
Comment (Kilo59) [inline great_expectations/expectations/regex_based_column_map_expectation.py]: This appears broken.

`num_to_str()` is not defined

File: great_expectations/expectations/set_based_column_map_expectation.py [hunk 4]
```diff
@@ -225,7 +225,7 @@ def _prescriptive_renderer(
             else:
                 template_str = "values must match this set: $set_"
             if params["mostly"] is not None:
-                params["mostly_pct"] = num_to_str(
+                params["mostly_pct"] = num_to_str(  # FIXME
```
Comment (Kilo59) [inline great_expectations/expectations/set_based_column_map_expectation.py]: This appears broken.

`num_to_str()` is not defined

File: great_expectations/expectations/regex_based_column_map_expectation.py [hunk 5]
```diff
@@ -189,7 +189,7 @@ def _atomic_prescriptive_template(
         else:
             template_str = "values must match this regular expression: $regex"
             if params["mostly"] is not None:
-                params_with_json_schema["mostly_pct"]["value"] = num_to_str(
+                params_with_json_schema["mostly_pct"]["value"] = num_to_str(  # FIXME
```
Comment (Kilo59) [inline great_expectations/expectations/regex_based_column_map_expectation.py]: This appears broken.

`num_to_str()` is not defined

File: great_expectations/expectations/row_conditions.py (hunk 1 above)
Comment (Kilo59) [inline great_expectations/expectations/row_conditions.py:160]:
```suggestion
) -> ColumnElement:
```

File: great_expectations/expectations/core/expect_column_values_to_match_like_pattern.py [hunk 6]
```diff
@@ -123,11 +123,9 @@ def _prescriptive_renderer(
...
     ) -> None:
         runtime_configuration = runtime_configuration or {}
-        include_column_name = (
-            False if runtime_configuration.get("include_column_name") is False else True
-        )
-        styling = runtime_configuration.get("styling")
-        params = substitute_none_for_missing(
+        _ = False if runtime_configuration.get("include_column_name") is False else True
+        _ = runtime_configuration.get("styling")
+        params = substitute_none_for_missing(  # noqa: F841 # unused
```
Comment (Kilo59) [inline great_expectations/expectations/core/expect_column_values_to_match_like_pattern.py:128]: A lot of the changes in this PR are for unused variables.
For some variables, I renamed them to `_` and others I just ignore the warning.

If the variable name expresses some useful information I ignore the error and keep the existing variable name.

But for `styling` and `include_column_name` those details are self-evident by the calls to the dictionary,

File: great_expectations/expectations/regex_based_column_map_expectation.py (hunk 3 above)
Comment (Kilo59) [reply great_expectations/expectations/regex_based_column_map_expectation.py]: Fixed

File: great_expectations/expectations/set_based_column_map_expectation.py (hunk 4 above)
Comment (Kilo59) [reply great_expectations/expectations/set_based_column_map_expectation.py]: Fixed

File: great_expectations/expectations/regex_based_column_map_expectation.py (hunk 5 above)
Comment (Kilo59) [reply great_expectations/expectations/regex_based_column_map_expectation.py]: Fixed

File: great_expectations/expectations/set_based_column_map_expectation.py [hunk 7]
```diff
@@ -29,6 +29,7 @@
     RendererValueType,
 )
 from great_expectations.render.util import (
+    num_to_str,
```
Comment (Kilo59) [inline great_expectations/expectations/set_based_column_map_expectation.py:32]: Added `num_to_str` import to fix bug

File: great_expectations/expectations/regex_based_column_map_expectation.py [hunk 8]
```diff
@@ -30,6 +30,7 @@
     RendererValueType,
 )
 from great_expectations.render.util import (
+    num_to_str,
```
Comment (Kilo59) [inline great_expectations/expectations/regex_based_column_map_expectation.py:33]: Added `num_to_str` import to fix bug

File: setup.cfg [hunk 9]
```diff
@@ -22,7 +22,7 @@ exclude = .git,
     versioneer*,
     examples/*,
     # TODO: remove the items below and fix linting issues
-    great_expectations/expectations, # 122 warnings
+    great_expectations/expectations/metrics, # 57 warnings
```
Comment (Kilo59) [inline setup.cfg:25]: We still have `expectations/metrics` to deal with.

File: great_expectations/expectations/core/expect_select_column_values_to_be_unique_within_record.py (hunk 2 above)
Comment (Kilo59) [reply great_expectations/expectations/core/expect_select_column_values_to_be_unique_within_record.py]: I've suppressed the warning, retained the FIXME comment, and will be creating a bug-ticket to fix this.

File: setup.py [hunk 10]
```diff
@@ -82,7 +82,9 @@ def get_extras_require():
     "cmdclass": versioneer.get_cmdclass(),
     "install_requires": required,
     "extras_require": get_extras_require(),
-    "packages": find_packages(exclude=["contrib*", "docs*", "tests*", "examples*", "scripts*"]),
+    "packages": find_packages(
+        exclude=["contrib*", "docs*", "tests*", "examples*", "scripts*"]
+    ),
```
Comment (Kilo59) [inline setup.py]: `setup.py` PR change got through our contrib CI without being properly formated.

Forcing this change in order to pass CI

Comment (Kilo59) [reply setup.py]: See https://github.com/great-expectations/great_expectations/pull/6756

File: great_expectations/expectations/core/expect_column_kl_divergence_to_be_less_than.py [hunk 11]
```diff
@@ -366,7 +366,7 @@ def get_validation_dependencies(
 
         return validation_dependencies
 
-    def _validate(
+    def _validate(  # noqa: C901 - 29
```
Comment (NathanFarmer) [inline great_expectations/expectations/core/expect_column_kl_divergence_to_be_less_than.py:369]: Would prefer that this says something about "too complex" or similar for people who don't know this code.

File: great_expectations/expectations/core/expect_column_kl_divergence_to_be_less_than.py [hunk 12]
```diff
@@ -528,7 +528,7 @@ def _validate(
             if "tail_weights" in partition_object:
                 partition_tail_weight_holdout = np.sum(partition_object["tail_weights"])
             else:
-                partition_tail_weight_holdout = 0
+                partition_tail_weight_holdout = 0  # noqa: F841 # unused
```
Comment (NathanFarmer) [inline great_expectations/expectations/core/expect_column_kl_divergence_to_be_less_than.py:531]: How is this unused but the one on line 529 is not?

File: great_expectations/expectations/core/expect_column_kl_divergence_to_be_less_than.py [hunk 13]
```diff
@@ -1152,7 +1152,7 @@ def _prescriptive_renderer(
         include_column_name = (
             False if runtime_configuration.get("include_column_name") is False else True
         )
-        styling = runtime_configuration.get("styling")
+        _ = runtime_configuration.get("styling")
```
Comment (NathanFarmer) [inline great_expectations/expectations/core/expect_column_kl_divergence_to_be_less_than.py:1155]: You can just remove this line, but this also fine. `styling` has been removed in the new-style renderers.

File: great_expectations/expectations/core/expect_select_column_values_to_be_unique_within_record.py [hunk 14]
```diff
@@ -177,9 +175,9 @@ def _prescriptive_renderer(
         )
 
         if params["mostly"] is not None and params["mostly"] < 1.0:
-            params_with_json_schema["mostly_pct"]["value"] = num_to_str(
-                params["mostly"] * 100, precision=15, no_scientific=True
-            )
+            params_with_json_schema["mostly_pct"][  # noqa: F821 # FIXME: this breaks
```
Comment (NathanFarmer) [inline great_expectations/expectations/core/expect_select_column_values_to_be_unique_within_record.py:178]: What does "this breaks" mean?

**Review by NathanFarmer — APPROVED** [review]
I am capturing the cleanup of these broken/legacy renderers in an epic, so ignoring the issues is fine for now.

**Pull request merged**